cmd/argocd/commands/admin/settings_rbac.go

```diff
@@ -194,24 +196,45 @@ argocd admin settings rbac can someuser create application 'default/app' --defau
 }
 
 // NewRBACValidateCommand returns a new rbac validate command
-func NewRBACValidateCommand() *cobra.Command {
+func NewRBACValidateCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
 	var (
 		policyFile string
 	)
 
 	var command = &cobra.Command{
-		Use:   "validate --policy-file=POLICYFILE",
+		Use:   "validate --policy-file=POLICYFILE [--namespace=NAMESPACE]",
 		Short: "Validate RBAC policy",
 		Long: `
 Validates an RBAC policy for being syntactically correct. The policy must be
 a local file, and in either CSV or K8s ConfigMap format.
 `,
 		Run: func(c *cobra.Command, args []string) {
-			if policyFile == "" {
+			if len(args) > 0 {
+				c.HelpFunc()(c, args)
+				log.Fatalf("too many arguments")
+			}
+
+			namespace, nsOverride, err := clientConfig.Namespace()
+			if err != nil {
+				log.Fatalf("could not create k8s client: %v", err)
+			}
```

Comment:
I think it would... But I think I like that better than having the API server pass back the whole ConfigMap. The RBAC CM seems like a somewhat sensitive resource to me.

Maybe we could just expand the error messages a bit to explain that you need kubectl access to the cm in order to use this feature.